Tasks/GoTool/gotool.ts

```diff
@@ -48,24 +48,26 @@ async function acquireGo(version: string): Promise<string> {
     //
     // Download - a tool installer intimately knows how to get the tool (and construct urls)
     //
-    version = toolLib.cleanVersion(version);
     let fileName: string = getFileName(version);
     let downloadUrl: string = getDownloadUrl(fileName);
     let downloadPath: string = null;
     try {
         downloadPath = await toolLib.downloadTool(downloadUrl);
     } catch (error) {
         tl.debug(error);
-        throw (tl.loc("VersionNotSupported", version));
+        throw (tl.loc("FailedToDownload", version, error));
     }
 
+    //make sure agent version is latest then 2.115.0
+    tl.assertAgent('2.115.0');
```

Comment:
Better to define such hard coded values (2.115.0) on the top as a 'const' variable, so that we don't have to search for this in code in future when we need to modify the value.

Reply:
This should be fine, as these value doesn't change

Reply:
We do this very frequently